include/rellic/AST/GenerateAST.h

```diff
@@ -61,7 +62,8 @@ class GenerateAST : public llvm::AnalysisInfoMixin<GenerateAST> {
                                     llvm::ConstantInt *c);
 
   z3::expr GetOrCreateEdgeCond(llvm::BasicBlock *from, llvm::BasicBlock *to);
-  z3::expr GetOrCreateReachingCond(llvm::BasicBlock *block);
+  z3::expr GetReachingCond(llvm::BasicBlock *block);
+  void CreateReachingCond(llvm::BasicBlock *block);
 
   clang::Expr *ConvertExpr(z3::expr expr);
```

Comment:
Have you considered using the existing code in `Z3ConvVisitor` directly?

Reply:
`Z3ConvVisitor` converts each expression recursively, and simplifying the resulting expression causes a mess when trying to recover where the original components came from (see the Z3->C fiasco). I wanted a very simple conversion procedure, where I always have a 1:1 correspondence between Z3 variables and LLVM values (almost, `switch` conditions are slightly different)

Reply:
Fair enough!